src/test/crimson/test_messenger_recovery.cc

```diff
@@ -0,0 +1,636 @@
+           : server_policy(srv_policy),
+             client_policy(cli_policy),
+             dispatcher(false, this),
+             servers(servers),
+             clients(clients) {
+
+           for (int i = 0; i < random_num; i++) {
+             bufferlist bl;
+             boost::uniform_int<> u(32, max_message_len);
+             uint64_t value_len = u(rng);
+             bufferptr bp(value_len);
+             bp.zero();
+             for (uint64_t j = 0; j < value_len-sizeof(i); ) {
+               memcpy(bp.c_str()+j, &i, sizeof(i));
+               j += 4096;
+             }
+
+             bl.append(bp);
+             rand_data.push_back(bl);
+           }
+         }
+
+         crimson::net::ConnectionRef _get_random_connection() {
+           while (dispatcher.get_pending() > max_in_flight) {
+             usleep(500);
+           }
+           boost::uniform_int<> choose(0, available_connections.size() - 1);
+           int index = choose(rng);
+           std::map<crimson::net::ConnectionRef,
+             std::pair<crimson::net::MessengerRef, crimson::net::MessengerRef>>::iterator i
+             = available_connections.begin();
+           for (; index > 0; --index, ++i) ;
+           return i->first;
+         }
+
+         bool can_create_connection() {
+           return available_connections.size() < max_connections;
+         }
+
+         seastar::future<> generate_connection() {
+           if (!can_create_connection()) {
+             return seastar::now();
+	   }
+           crimson::net::MessengerRef server, client;
+           {
+             boost::uniform_int<> choose(0, available_servers.size() - 1);
+             int index = choose(rng);
+             std::set<crimson::net::MessengerRef>::iterator i
+               = available_servers.begin();
+             for (; index > 0; --index, ++i) ;
+             server = *i;
+           }
+           {
+             boost::uniform_int<> choose(0, available_clients.size() - 1);
+             int index = choose(rng);
+             std::set<crimson::net::MessengerRef>::iterator i
+               = available_clients.begin();
+             for (; index > 0; --index, ++i) ;
+             client = *i;
+           }
+
+
+           std::pair<crimson::net::MessengerRef, crimson::net::MessengerRef>
+	     connected_pair;
+           {
+	       crimson::net::ConnectionRef conn = client->connect(
+		                                    server->get_myaddr(),
+		                                    entity_name_t::TYPE_OSD);
+             connected_pair = std::make_pair(client, server);
+             available_connections[conn] = connected_pair;
+           }
+           return seastar::now();
+         }
+
+	 seastar::future<> random_op (const uint64_t& iter) {
+	   return seastar::do_with(iter, [this] (uint64_t& iter) {
+	     return seastar::do_until([&] {
+	       return iter == 0; },
+	     [&, this] {
+	       if (!(iter % 10)) {
+	         logger().info("Op {} : ", iter);
```

Comment:
Please add log prefix to help group and locate the logs and for better readability.

In this case, "SyntheticWorkload::random_op(): "